packages/std/src/math.rs

```diff
@@ -89,18 +90,35 @@ impl FromStr for Decimal {
                     .ok_or_else(|| generic_err("Cannot compute fractional factor"))?;
 
                 let whole_as_atomics = whole
-                    .checked_mul(DECIMAL_FRACTIONAL.u128())
+                    .checked_mul(DECIMAL_FRACTIONAL)
                     .ok_or_else(|| generic_err("Value too big"))?;
                 let atomics = whole_as_atomics
                     .checked_add(fractional * fractional_factor)
                     .ok_or_else(|| generic_err("Value too big"))?;
-                Ok(Decimal(Uint128(atomics)))
+                Ok(Decimal(atomics))
             }
             _ => Err(generic_err("Unexpected number of dots")),
         }
     }
 }
 
+impl fmt::Display for Decimal {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let whole = (self.0) / DECIMAL_FRACTIONAL;
+        let fractional = (self.0) % DECIMAL_FRACTIONAL;
+
+        if fractional == 0 {
+            write!(f, "{}", whole)
+        } else {
+            let fractional_string = format!("{:018}", fractional);
+            f.write_str(&whole.to_string())?;
+            f.write_char('.')?;
+            f.write_str(fractional_string.trim_end_matches('0'))?;
```

Comment:
cool, so easy to remove the trailing '0'
Think this is better than the version in the sdk. And in one afternoon :medal_sports: